pentaho-js/src/main/javascript/cdf-legacy/components/Pentaho.Analyzer.js

```diff
@@ -55,6 +56,12 @@
         // process params and update options
         $.map(this.parameters, function(k) {
             options[k[0]] = k.length == 3 ? k[2] : Dashboards.getParameterValue(k[1]);
+            if(typeof myself.dateFormats !== 'undefined' && myself.dateFormats[k[0]]){
```

Comment:
I think we can explicitly set dateFormats property in this component as an empty object, skipping this extra check, and documenting it a bit, something like:
``` javascript
// parameterName / dateFormat Map - describes the dateFormats of the associated parameters
// eg. parameterName: "YYYY-MM-DD HH-mm-ss.0"
dateFormats: {},
```